OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbySavedSettingsBrowserLogic.cs

```diff
@@ -0,0 +1,351 @@
+							return false;
+
+						return true;
+					});
+			};
+
+			var deleteButton = panel.Get<ButtonWidget>("DELETE_BUTTON");
+			deleteButton.IsDisabled = () => selectedSave == null;
+			deleteButton.OnClick = () =>
+			{
+				ConfirmationDialogs.ButtonPrompt(
+					title: "Delete selected lobby settings save?",
+					text: "Delete '{0}'?".F(Path.GetFileNameWithoutExtension(selectedSave)),
+					onConfirm: () =>
+					{
+						Delete(selectedSave);
+
+						if (!saves.Any() && !isSavePanel)
+						{
+							Ui.CloseWindow();
+						}
+						else
+							SelectFirstVisible();
+					},
+					confirmText: "Delete",
+					onCancel: () => { });
+			};
+
+			LoadExistingSaves(newTemplate, savedTemplate);
+			SelectFirstVisible();
+		}
+
+		void SelectFirstVisible()
+		{
+			Select(isSavePanel ? null : saves.FirstOrDefault());
+		}
+
+		void Select(string savePath)
+		{
+			selectedSave = savePath;
+			if (isSavePanel)
+				saveTextField.Text = savePath == null ? defaultSaveFilename :
+					Path.GetFileNameWithoutExtension(savePath);
+		}
+
+		void LoadExistingSaves(ScrollItemWidget newTemplate, ScrollItemWidget savedTemplate)
+		{
+			saveList.RemoveChildren();
+
+			if (isSavePanel)
+			{
+				var item = ScrollItemWidget.Setup(newTemplate,
+					() => selectedSave == null || !saves.Contains(SettingsFilePath(saveTextField.Text)),
+					() => { });
+				saveList.AddChild(item);
+			}
+
+			if (Directory.Exists(baseSavePath))
+			{
+				var saveFiles = Directory.GetFiles(baseSavePath, "*.yaml", SearchOption.AllDirectories)
+					.OrderByDescending(p => File.GetLastWriteTime(p))
+					.ToList();
+
+				foreach (var savePath in saveFiles)
+				{
+					saves.Add(savePath);
+
+					// Create the item manually so the click handlers can refer to itself
+					// This simplifies the rename handling (only needs to update ItemKey)
+					var item = savedTemplate.Clone() as ScrollItemWidget;
+					item.ItemKey = savePath;
+					item.IsVisible = () => true;
+					item.OnClick = () => Select(item.ItemKey);
+					if (isSavePanel)
+					{
+						item.IsSelected = () => saveTextField.Text == Path.GetFileNameWithoutExtension(item.ItemKey);
+						item.OnDoubleClick = Save;
+					}
+					else
+					{
+						item.IsSelected = () => selectedSave == item.ItemKey;
+						item.OnDoubleClick = Load;
+					}
+
+					var title = Path.GetFileNameWithoutExtension(savePath);
+					var label = item.Get<LabelWithTooltipWidget>("TITLE");
+					WidgetUtils.TruncateLabelToTooltip(label, title);
+
+					var date = File.GetLastWriteTime(savePath).ToString("yyyy-MM-dd HH:mm:ss");
+					item.Get<LabelWidget>("DATE").GetText = () => date;
+
+					saveList.AddChild(item);
+				}
+			}
+		}
+
+		void Save()
+		{
+			var globalSettings = orderManager.LobbyInfo.GlobalSettings.Serialize();
+			var options = globalSettings.Value.ToDictionary()["Options"];
+			var unlockedOptions = options.Nodes.FindAll(optionYaml => optionYaml.Value.ToDictionary()["IsLocked"].Value == "False");
```

Comment:
In lieu of strongly-typed data, we should probably do a case-insensitive comparison since, I _believe_, the MiniYaml loader supports `False` and `false`, for example.